yaml/_dumper.ts

```diff
@@ -578,13 +577,11 @@ function writeScalar(
         return `'${string.replace(/'/g, "''")}'`;
       case STYLE_LITERAL:
         return `|${blockHeader(string, state.indent)}${
-          dropEndingNewline(
-            indentString(string, indent),
-          )
+          trimTrailingNewline(indentString(string, indent))
```

Comment:
Anything wrong with this?
```suggestion
          indentString(string, indent).trimEnd()
```

Reply:
I think this will also trim whitespaces. Not sure if that is a problem though.

Reply:
Yeah, it'll trim all whitespaces, and I think that's fine. @kt3k, see any issues with that?

Reply:
That will break the case when the string has white spaces at the end.

(The 2nd case of https://github.com/denoland/deno_std/pull/5341)

Reply:
I know, but in what cases is that an issue? Genuine question.

Reply:
When the user's input has trailing whitespace, then that part needs to be serialized as is. Trimming that whitespace means the user loses some data while serializing.

Reply:
Oh, yes, that's right. Obvious once you say it 😆